zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/AppendOutputRunner.java

```diff
@@ -0,0 +1,112 @@
+package org.apache.zeppelin.interpreter.remote;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This thread sends paragraph's append-data
+ * periodically, rather than continously, with
+ * a period of BUFFER_TIME_MS. It handles append-data
+ * for all paragraphs across all notebooks.
+ */
+public class AppendOutputRunner implements Runnable {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(AppendOutputRunner.class);
+  private static RemoteInterpreterProcessListener listener;
+
+  private static final Long BUFFER_TIME_MS = new Long(100);
+  private static final Long SAFE_PROCESSING_TIME = new Long(10);
+  private static final Long SAFE_PROCESSING_STRING_SIZE = new Long(100000);
+
+  private static final BlockingQueue<AppendOutputBuffer> QUEUE =
+      new LinkedBlockingQueue<AppendOutputBuffer>();
+
+  @Override
+  public void run() {
+
+    while (true) {
```

Comment:
I have a doubt. Do we want to remove this while loop and instead use ScheduledExecutorService to spawn a thread everytime, which will substitute this looping? I didn't do it because it would need over-head of a thread creation very frequently (may use lot of CPU and GC overhead). I was thinking that the job of ScheduledExecutorService (or CheckAppendOutputRunner ) is just to verify that AppendOutputRunner thread is running, and if not, then start it. That way we save overhead of spawning threads. Should I change this?